esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -2894,7 +2906,29 @@ void CWsWorkunitsEx::getWsWuResult(IEspContext &context, const char *wuid, const
     else
         rs.setown(resultSetFactory->createNewResultSet(result, wuid));
     if (!filterBy || !filterBy->length())
+    {
+        unsigned __int64 resultSz;
+
+        if (0 == logicalName.length()) // could be a workunit owned file (OUTPUT, THOR)
+            result->getResultFilename(logicalName);
+        if (logicalName.length())
+        {
+            Owned<IDistributedFile> df = lookupLogicalName(context, logicalName.str());
+            if (!df)
+                throw makeStringExceptionV(ECLWATCH_FILE_NOT_EXIST, "Cannot find file %s.", logicalName.str());
+            resultSz = df->getDiskSize(true, false);
+        }
+        else
+            resultSz = result->getResultRawSize(nullptr, nullptr);
```

Comment:
Why do you prefer df->getDiskSize() more than result->getResultRawSize()? Could df->getDiskSize() return the size of whole file, but, the result is a part of the file?

Reply:
@wangkx - getResultRawSize will return 0, if it's a disk based result (raw result is for a workunit result only)

Reply:
I see.